auxx/src/Mode.hs

```diff
@@ -147,7 +147,7 @@ instance HasSlogGState AuxxContext where
 instance HasJsonLogConfig AuxxContext where
     jsonLogConfig = acRealModeContext_L . jsonLogConfig
 
-instance (HasConfiguration, HasInfraConfiguration, MonadSlotsData ctx AuxxMode)
+instance (HasConfiguration, HasNtpConfiguration, MonadSlotsData ctx AuxxMode)
```

Comment:
Do you think it would be feasible to stop using the `HasNtpConfiguration` style and just take a record instead? Typically the problem with doing such a change is that the configuration is used in instances, but if we could make it not needed in the instances...

Reply:
I think that's feasible :). Previously the `HasNtpConfiguration` was used to provide the configuration for ntp slotting that we removed.

Reply:
@avieth it is possible to eliminate it from `cardano-sl`, but then there is `HasConfigurations` which is using it all over the place in the wallet and explorer.
Furthermore, `withConfigurations` is the function that reads the yaml config, including `ntpConfiguration` part, so the ntp config is coupled with all other configs.

I advanced to the point where ntp configuration is passed explicitly in `cardano-sl` and the `cardano-sl-explorer` compiles, but still the `HasNtpConfiguration` is used in `HasConfigurations` (beacuse of `withConfigurations`).  This means that the functions `runNodeReal`, `bracketNodeResources` (and its friends: `allocateNodeResources`, `allocateNodeContext`), get `NtpConfiguration` explicitly.  I was able to remove the `HasNtpConfiguration` constraint from various instances in `cardano-sl` and `cardano-sl-networking`.  The only place `HasNtpConfiguration` is really needed is in the wallet.

To conclude: I don't think we can simply remove `HasNtpConfiguration` without refactoring `withConfigurations` which I didn't want to touch for now.  The question I have, should I intrude into wallet so that the ntp configuration is passed explicitly?

Another suggestion I have is that we probably should have `NtpNotConfigured` constructor for `NtpConfiguration` (making `NtpConfiguration ~ Maybe NtpConfiguration`).  Simply because ntp is not required in all cases (mostly experiments/tests or the explorer app).  This will force us to add `NtpSyncUnavailable` to `NtpStatus` (to keep all the functions total).

Reply:
I pushed the changes in a separate branch (last commit): https://github.com/input-output-hk/cardano-sl/tree/coot/csl-2346-remove-HasNtpConfiguration (I updated the `wallet` and `new-wallet`, with some more `HasNtpConfiguration` so that it compiles :/)

Reply:
Relevant refactoring here https://github.com/input-output-hk/cardano-sl/pull/2590
That patch removes all `HasXConfiguration` from the diffusion layer, but of course leaves it in place for other areas because it's heavily entrenched, particularly the dependences of slotting definitions on the `k` parameter from `HasProtocolConstants`, which determines the number of slots in an epoch and thereby makes it possible to get `Num`, `Enum`, `Bounded` instances for local slot indices.

Anyway, you're right it would be a bit of a job to fix this now, so I've made a ticket and we can do it later https://iohk.myjetbrains.com/youtrack/issue/CSL-2370